### PR TITLE
multiple motion fixes and improvements

### PR DESCRIPTION
**Pull request by SammyRamone**
## Proposed changes
- Sometimes the IK had issues on start up as something was not initialized correctly. fix ik issues due to missing setting of the state to default
- configs for deep quintic
- add methods to python walk wrapper to check if a certain cmd_vel is kinematicially possible for a robot
- add collision checking to moveit bindings
- reduce spam from odometry on launch
- use event executor for walk and odometry nodes
- fix and cleanup launches
- overall spam reduction

## Necessary checks
- [ ] Update package version
- [ ] Run `catkin build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot
- [ ] Put the PR on our Project board

### CODE REVIEW

**Comment by timonegk** [comment]
The change itself looks fine, but this pull request contains some stuff that has nothing to do with it. Should it be merged as well or was it an accident?

**Comment by SammyRamone** [comment]
> The change itself looks fine, but this pull request contains some stuff that has nothing to do with it. Should it be merged as well or was it an accident?

Yeah the branch became my WIP branch and has a lot of stuff now :sweat_smile: I will clean it up and make it merge ready when I am finished with the current features

**Comment by SammyRamone** [comment]
currently the odom frame is not published. I don't really know why, since the broadcast is called

**Review by timonegk — APPROVED** [review]
I think this should be merged as soon as possible. It builds on my machine but I am still having trouble launching the complete motion stack. However, I suppose it is easier to fix these issues later in a separate pull request, especially if it works on your machine, as this one gets too large to do any productive work and the current state on master is not better, either.

**Comment by SammyRamone** [comment]
Currently the odometry is partially broken on this branch, I want to fix this before merging

**Comment by SammyRamone** [comment]
additionally there is also currently the following warning, that should probably be resolved, but this maybe needs to be done in the bio_ik git 
[DynupNode-16] Warning: class_loader.impl: SEVERE WARNING!!! A namespace collision has occurred with plugin factory for class bio_ik_kinematics_plugin::BioIKKinematicsPlugin. New factory will OVERWRITE existing one. This situation occurs when libraries containing plugins are directly linked against an executable (the one running right now generating this message). Please separate plugins out into their own library or just don't link against the library and use either class_loader::ClassLoader/MultiLibraryClassLoader to open.

File: bitbots_moveit_bindings/bitbots_moveit_bindings/__init__.py [hunk 1]
```diff
@@ -2,33 +2,70 @@
...
+    global ik_fk_state, collision_state, current_state
     serialized_parameters = []
     for parameter in parameters:
         serialized_parameters.append(serialize_message(parameter))
-    bindings = BitbotsMoveitBindings(serialized_parameters)
+    if type=="ikfk":
+        ik_fk_state = BitbotsMoveitBindings(serialized_parameters)
+    elif type=="collsion":
+        collision_state = BitbotsMoveitBindings(serialized_parameters)
+    elif type=="current":
```
Comment (timonegk) [inline bitbots_moveit_bindings/bitbots_moveit_bindings/__init__.py:29]:
```suggestion
    if type == "ikfk":
        ik_fk_state = BitbotsMoveitBindings(serialized_parameters)
    elif type == "collision":
        collision_state = BitbotsMoveitBindings(serialized_parameters)
    elif type == "current":
```

File: bitbots_moveit_bindings/src/bitbots_moveit_bindings.cpp [hunk 2]
```diff
@@ -187,19 +195,197 @@ class BitbotsMoveitBindings {
...
   m.def("initRos", &ros2_python_extension::initRos);
   py::class_<BitbotsMoveitBindings, std::shared_ptr<BitbotsMoveitBindings>>(m, "BitbotsMoveitBindings")
       .def(py::init<std::vector<py::bytes>>())
       .def("getPositionIK", &BitbotsMoveitBindings::getPositionIK)
-      .def("getPositionFK", &BitbotsMoveitBindings::getPositionFK);
+      .def("getPositionFK", &BitbotsMoveitBindings::getPositionFK)
+      .def("set_head_motors", &BitbotsMoveitBindings::setHeadMotors, "Set the current pan and tilt joint values [radian]", py::arg("pan"), py::arg("tilt"))
+      .def("set_joint_states", &BitbotsMoveitBindings::setJointStates, "Set the current joint states")
+      .def("check_collision", &BitbotsMoveitBindings::checkCollision, "Returns true if the head collides, else false")
+      .def("get_joint_states", &BitbotsMoveitBindings::getJointStates, "Set the current joint states");
```
Comment (timonegk) [inline bitbots_moveit_bindings/src/bitbots_moveit_bindings.cpp:390]:
```suggestion
      .def("get_joint_states", &BitbotsMoveitBindings::getJointStates, "Get the current joint states");
```

**Comment by timonegk** [comment]
My problem was on my side, I tested again and now it works, except of the warning that you mentioned. But I can look into that, I don't think it's related to this PR. For the odometry problem: That's what I mean, I think this can still be merged before because the current master is broken anyway, and by merging it, we can continue the work in other areas that don't need a working odometry.

**Comment by SammyRamone** [comment]
I already found out what the issue is. the time for the transform is not set. Just give me a few more minutes, maybe I can come up with a fix

**Comment by SammyRamone** [comment]
Odometry now works again. In my opinion it can be merged.
Please have a look at the recent changes and merge if its okay for you @timonegk

**Comment by timonegk** [comment]
Perfect, I didn't know you were so close to finding a fix

**Pull request merged**